bonsai-core/src/main/scala/com/stripe/bonsai/IndexedBitSet.scala

```diff
@@ -160,20 +244,53 @@ object IndexedBitSet {
     }
 
   /**
-   * TODO: maybe tom can explain this?
+   * Returns the rank of the i-th bit in a 32-bit word.
    */
-  def rankWord(word: Int, i: Int): Int = {
+  private[bonsai] def rankWord(word: Int, i: Int): Int = {
     val mask = ~(-1L << ((i + 1))).toInt
     java.lang.Integer.bitCount(word & mask)
   }
 
   /**
+   * Returns the index of the i-th set bit in a 32-bit word.
+   *
+   * @param word a 32-bit word to search
+   * @param rank a value in [1,32] (inclusive)
+   */
+  private[bonsai] def selectWord(word: Int, rank: Int): Int = {
```

Comment:
Assuming `bitCount` is doing something like [this](http://stackoverflow.com/a/109025), I imagine an optimization would be to bail out early when you are a bit below the target and do some extra shifts to find it via the naïve method. But you'd have to profile it to figure out what the optimal level would be.